api/get_mass_migration_commitment_proof.go

```diff
@@ -0,0 +1,144 @@
+package api
+
+import (
+	"github.com/Worldcoin/hubble-commander/commander/executor"
+	"github.com/Worldcoin/hubble-commander/encoder"
+	"github.com/Worldcoin/hubble-commander/models"
+	"github.com/Worldcoin/hubble-commander/models/dto"
+	"github.com/Worldcoin/hubble-commander/models/enums/batchtype"
+	"github.com/Worldcoin/hubble-commander/storage"
+	"github.com/Worldcoin/hubble-commander/utils/merkletree"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+)
+
+var getMassMigrationCommitmentProofAPIErrors = map[error]*APIError{
+	storage.AnyNotFoundError: NewAPIError(50004, "mass migration commitment inclusion proof not found"),
+}
+
+func (a *API) GetMassMigrationCommitmentProof(batchID models.Uint256, commitmentIndex uint8) (*dto.MassMigrationCommitmentProof, error) {
+	if !a.cfg.EnableProofMethods {
+		return nil, errProofMethodsDisabled
+	}
+	commitmentInclusionProof, err := a.unsafeGetMassMigrationCommitmentProof(batchID, commitmentIndex)
+	if err != nil {
+		return nil, sanitizeError(err, getMassMigrationCommitmentProofAPIErrors)
+	}
+	return commitmentInclusionProof, nil
+}
+
+func (a *API) unsafeGetMassMigrationCommitmentProof(
+	batchID models.Uint256,
+	commitmentIndex uint8,
+) (*dto.MassMigrationCommitmentProof, error) {
+	batch, err := a.storage.GetBatch(batchID)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	commitmentID := models.CommitmentID{
+		BatchID:      batchID,
+		IndexInBatch: commitmentIndex,
+	}
+
+	commitments, err := a.storage.GetCommitmentsByBatchID(batchID, batchtype.MassMigration)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	unsortedMassMigrations, err := a.storage.GetMassMigrationsByCommitmentID(commitmentID)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	// TODO remove when new primary key for transactions with transaction index is implement
+	txQueue := executor.NewTxQueue(models.MassMigrationArray(unsortedMassMigrations))
+	massMigrations := txQueue.PickTxsForCommitment().ToMassMigrationArray()
+
+	serializedMassMigrations, err := encoder.SerializeMassMigrations(massMigrations)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	commitment, err := a.storage.GetTxCommitment(&commitmentID)
```

Comment:
We can use `commitment := commitments[commitmentID.IndexInBatch].(*models.TxCommitment)` instead of that query.